### PR TITLE
Ben/auto and detect

### PR DESCRIPTION
**Pull request by taitmate38**
A fairly chonky merge, doing a couple different things. Result of merge between multiple PRs, relating to automation of flow start/stop, configurability for cryo-therms, adding a second control loop for LOX Gems, and a packet-counter packet that is sent down periodically.
Automation: `automation.h`, `automation.cpp` & relevant calls were completely overhauled to instead use a rudimentary event-based system for executing automated sequences. 
Detection: Nolan added an automatic detection that will detect when a propellant has run out on a given side based on the spike in pressure that is usually observed.

from Vamshi's PR:
Solves: https://trello.com/c/Vt5EYNsW/203-cryo-therm-heater-control-loop
Use 2 Cryo Therms for lox adapter tree.
Two heater control loops use these cryo thermocouples as inputs.
fix small bugs from @nehading's latest PR re packet counter.
Added appropriate packet information to the design sheet.

### CODE REVIEW

File: lib/ducer/ducer.h [hunk 1]
```diff
@@ -107,7 +107,8 @@ namespace Ducers {
         data[i] = interpolateLow(_adcs[_adcIndices[i]]->readData(_adcChannels[i]));
         i++;
       } else {
-        data[i] = interpolateHigh(_adcs[_adcIndices[i]]->readData(_adcChannels[i]));
+        // data[i] = interpolateHigh(_adcs[_adcIndices[i]]->readData(_adcChannels[i]));
+        data[i] = _adcs[_adcIndices[i]]->readData(_adcChannels[i]);
```
Comment (taitmate38) [inline lib/ducer/ducer.h:111]: Was having issues with on-board interpolation for high pressure on 2/13/21. Nolan is looking into them, for now using the interpolation on the ground station side

**Comment by taitmate38** [comment]
[Startup & Shutdown Automation](https://trello.com/c/6olXYIGq/179-startup-shutdown-automation)

File: src/E1_coldflow/config.h [hunk 2]
```diff
@@ -24,16 +27,15 @@ const int numAnalogThermocouples = 1;
...
 int thermAdcChannels[numAnalogThermocouples] = {2};
 
-const int numPressureTransducers = 5;
-int ptAdcIndices[numPressureTransducers] = {0, 0, 0, 0, 1};
-int ptAdcChannels[numPressureTransducers] = {0, 1, 2, 3, 0};
-int ptTypes[numPressureTransducers] = {1, 1, 1, 1, 2};
+const int numPressureTransducers = 7;
+int ptAdcIndices[numPressureTransducers] = {0, 0, 0, 0, 1, 1, 1}; //not using 1-0 or 1-3
+int ptAdcChannels[numPressureTransducers] = {0, 1, 2, 3, 2, 1, 3};
+int ptTypes[numPressureTransducers] = {1, 1, 1, 1, 2, 1, 1};
```
Comment (fazerlicourice7) [inline src/E1_coldflow/config.h:33]: remove the comms w/ the broken channels. just slows things down

File: src/E1_coldflow/E1_coldflow.cpp [hunk 3]
```diff
@@ -140,10 +159,32 @@ void loop() {
...
+    
+
+    if (Solenoids::getLox5()) { //if LOX Main valve open
+      Serial.print("0: "); Serial.println(loxInjector);
+      Automation::detectPeak(loxInjector, 0);
+    }
+    if (Solenoids::getProp5()) { //if Prop Main valve open
+      Serial.print("1: "); Serial.println(propInjector);
+      Automation::detectPeak(propInjector, 1);
+    }
```
Comment (fazerlicourice7) [inline src/E1_coldflow/E1_coldflow.cpp]: This looks like it should be its own func in automation

File: src/E1_coldflow/config.h (hunk 2 above)
Comment (fazerlicourice7) [reply src/E1_coldflow/config.h:33]: nvm

Comment (fazerlicourice7) [reply src/E1_coldflow/config.h:33]: gs requires paket of certain size

File: src/E1_coldflow/E1_coldflow.cpp (hunk 3 above)
Comment (taitmate38) [reply src/E1_coldflow/E1_coldflow.cpp]: moved logic to Automation::detectPeaks()

**Review by fazerlicourice7 — APPROVED** [review]
cool. I'm merging!

**Pull request merged**